tests/test_measure_companions.py

```diff
@@ -0,0 +1,156 @@
+    fluxcal_factor = get_fluxcal_factor(direct_star_image, PHOT_METHOD, PHOT_ARGS, FLUX_OR_IRR)
+    zero_point = fluxcal_factor.ext_hdr.get('ZP', None)
+
+    # Generate Calibration Data
+    dataset_ct, ct_cal = mocks.create_mock_ct_dataset_and_cal_file(fwhm=50, n_psfs=20, cfam_name='3C', save_cal_file=True)
+    FpamFsamCal = mocks.create_mock_fpamfsam_cal(save_file=False)
+
+    print("Companion 1 AP Mag:", (-2.5 * np.log10(host_star_counts / 2) + zero_point))
+    print("Companion 2 AP Mag:", (-2.5 * np.log10(host_star_counts / 3) + zero_point))
+
+    # Generate PSF-subtracted Frame
+    psf_sub_frame = mocks.generate_psfsub_image_with_companions(
+        nx=200, ny=200, host_star_center=None, host_star_counts=host_star_counts,
+        psf_sub_scale=0.7, companion_xy=[(120, 80), (90, 130)],
+        companion_counts=[host_star_counts / 2, host_star_counts / 3],
+        companion_mags=[(-2.5 * np.log10(host_star_counts / 2) + zero_point),
+                        (-2.5 * np.log10(host_star_counts / 3) + zero_point)],
+        zero_point=zero_point, ct_cal=ct_cal, use_ct_cal=True, cor_dataset=dataset_ct,
+        FpamFsamCal=FpamFsamCal, blur_sigma=0.5, noise_std=1e-8, outdir=out_dir, platescale=0.0218
+    )
+
+    coron_data = mocks.generate_coron_dataset_with_companions(
+        n_frames=1, shape=(200, 200), companion_xy=[(120, 80), (90, 130)], companion_counts=[host_star_counts / 2, host_star_counts / 3],
+        host_star_counts=host_star_counts, roll_angles=[10.0], add_noise=True, noise_std=1e-8, outdir=out_dir
+    )
+
+    return direct_star_image, host_star_counts, fluxcal_factor, zero_point, dataset_ct, ct_cal, FpamFsamCal, psf_sub_frame, coron_data
+
+
+def test_measure_companions_wcs(out_dir):
+    """
+    Run the full test case for measuring companions.
+    """
+    direct_star_image, host_star_counts, fluxcal_factor, zero_point, dataset_ct, ct_cal, FpamFsamCal, \
+        psf_sub_frame, coron_data = generate_test_data(out_dir)
+    
+    # Get reference psf (it is near 6 lam/D), assume off-axis PSF with highest CT has negligible 
+    # effect from the mask. For now I am identifying it here and passing it in, but maybe we can 
+    # just pass in the whole dataset and pick it out using distance or highest CT.
+    reference_psf = ct_cal.data[0]
+    reference_psf = Image(data_or_filepath=reference_psf, pri_hdr=ct_cal.pri_hdr, ext_hdr=ct_cal.ext_hdr)
+
+    # Run Measurement
+    result_table = measure_companions.measure_companions(
+        psf_sub_image=psf_sub_frame, coronagraphic_dataset=coron_data,
+        phot_method='aperture',
+        ct_cal=ct_cal, ct_dataset=dataset_ct, FpamFsamCal=FpamFsamCal,
+        fluxcal_factor=fluxcal_factor,
+        forward_model=False, direct_star_image= direct_star_image,
+        reference_psf=reference_psf, output_dir = out_dir, verbose=True
+    )
+
+    print("\nResult Table:\n", result_table)
+    assert len(result_table) == 2, "Expected 2 measured companions"
```

Comment:
I assume we'll eventually want to check that the flux/magnitude we get is as expected.